### PR TITLE
remove superflous string formatting

### PR DESCRIPTION
**Pull request by Flamefire**
`"%s" % var` is the same as `str(var)` while the latter is faster due to less parsing, inserting etc.
In some places `var` was already a string so it can just be used.

### CODE REVIEW

File: easybuild/framework/easyconfig/format/format.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def parse(self, configobj):
         for key, value in self.supported.items():
             if key not in self.VERSION_OPERATOR_VALUE_TYPES:
                 raise EasyBuildError('Unsupported key %s in %s section', key, self.SECTION_MARKER_SUPPORTED)
-            self.sections['%s' % key] = value
+            self.sections[key] = value
```
Comment (boegel) [inline easybuild/framework/easyconfig/format/format.py:373]: This is not fully equivalent, are we sure we don't need `str(key)` here?

File: easybuild/tools/docs.py [hunk 2]
```diff
@@ -1199,7 +1199,7 @@ def avail_toolchain_opts_md(name, tc_dict):
     tc_items = sorted(tc_dict.items())
     table_values = [
         ['``%s``' % val[0] for val in tc_items],
-        ['%s' % val[1][1] for val in tc_items],
+        [val[1][1] for val in tc_items],
```
Comment (boegel) [inline easybuild/tools/docs.py:1202]: no need for `str(val[1][1])`?

File: test/framework/sandbox/easybuild/tools/module_naming_scheme/test_module_naming_scheme.py [hunk 3]
```diff
@@ -64,6 +64,6 @@ def det_module_symlink_paths(self, ec):
 
     def is_short_modname_for(self, modname, name):
         """
-        Determine whether the specified (short) module name is a module for software with the specified name.
+        Determine whether the specified (short) module name is a moduleq for software with the specified name.
```
Comment (boegel) [inline test/framework/sandbox/easybuild/tools/module_naming_scheme/test_module_naming_scheme.py:67]: typo introduced?

```suggestion
        Determine whether the specified (short) module name is a module for software with the specified name.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/framework/easyconfig/format/format.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/framework/easyconfig/format/format.py:373]: That question is actually a good reason **for** this change ;-)   
The change is correct: This line is never executed if the key is not also a key in `VERSION_OPERATOR_VALUE_TYPES` which are all strings and cannot reasonably be anything else.   
Alternatively: A non-string key doesn't make sense at this point.

File: easybuild/tools/docs.py (hunk 2 above)
Comment (Flamefire) [reply easybuild/tools/docs.py:1202]: `tc_dict` is dict mapping toolchain options to a tuple of `(default_value, description)`. And the description can only be string (and is for all examples I could find)

File: easybuild/framework/easyconfig/format/format.py (hunk 1 above)
Comment (boegel) [reply easybuild/framework/easyconfig/format/format.py:373]: OK, thanks for clarifying!

File: easybuild/tools/docs.py (hunk 2 above)
Comment (boegel) [reply easybuild/tools/docs.py:1202]: :+1:

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**